### PR TITLE
macOS SDK for Conda

### PR DESCRIPTION
**Pull request by maxscheurer**
As it says in the title.

### CODE REVIEW

File: dist_tools/conda/spline/meta.yaml [hunk 1]
```diff
@@ -29,13 +29,15 @@ requirements:
...
-        - pytest -s python/test/test_basic.py
+test:
+    imports:
+      - spline
+    requires:
+      - pytest
+    # source-files:
+    #   - python/test/test_basic.py
+    # commands:
+    #   - pytest -s python/test/test_basic.py
```
Comment (maxscheurer) [inline dist_tools/conda/spline/meta.yaml:40]: @Haydnspass the `test` section had to be out-dented 😁

File: dist_tools/conda/spline/build.sh
Comment (Haydnspass) [inline dist_tools/conda/spline/build.sh:1]: I don't understand this but I guess that's fine

Comment (maxscheurer) [reply dist_tools/conda/spline/build.sh:1]: Me neither... installing with `setup.py install` is deprecated anyways, so maybe a step in the right direction 😂

File: python/setup.py [hunk 2]
```diff
@@ -51,6 +50,10 @@ def build_extension(self, ext):
         cmake_args += ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir]
         if "CONDA_BUILD" in env:
             cmake_args += [f'-DPython_EXECUTABLE={env["PYTHON"]}']
+        sysroot = os.environ.get("CONDA_BUILD_SYSROOT", None)
+        import sys
```
Comment (Haydnspass) [inline python/setup.py:54]: Was there a reason to put the import here?

File: .github/workflows/build_upload_test.yaml [hunk 3]
```diff
@@ -16,7 +17,7 @@ jobs:
       matrix:
         include:
           # Windows2019 & VS 2019 supports 10.1+
-          - os: ubuntu-18.04
+          - os: ubuntu-latest
```
Comment (Haydnspass) [inline .github/workflows/build_upload_test.yaml:20]: I think I'd prefer pinning to 22.04, but I can change it later on

File: .github/workflows/build_upload_test.yaml [hunk 4]
```diff
@@ -2,6 +2,7 @@ name: build, upload and lazy test
 
 on:
   push:
+    branches: [master]
```
Comment (Haydnspass) [inline .github/workflows/build_upload_test.yaml:5]: Hm

**Pull request merged**